src/lib/qnio/iioapi.c

```diff
@@ -467,6 +470,14 @@ iio_open(const char *uri, const char *devid, uint32_t flags)
         return NULL;
     }
 
+    /* devid is used to compose a fixed-length filename string, so the
+     * string input size must be bounded */
+    if (strlen(devid) > MAX_DEV_ID_SZ - 1) {
+        nioDbg("device name cannot exceed %d characters\n", MAX_DEV_ID_SZ - 1);
+        errno = EINVAL;
+        return NULL;
+    }
+
```

Comment:
Thanks for the fix. On a related note, from the qemu code, we expect the user to only pass a vdisk UUID as the devid. Therefore there is a check in qemu vxhs.c to makes sure the vdisk-id passed is not > 36 chars (length of UUID).

libvxhs, in the spirit of being a more generic library, can allow > 36 chars for devid. So 255 looks to be a good number.